### PR TITLE
Widget

### LINKED ISSUE #27: Make a sub service of Cycling Norway into a Slippy Map/Widget
Similar issue #11, but not the same.

Have the Cycling Norway solution also work as a [Slippy Map](https://wiki.openstreetmap.org/wiki/Deploying_your_own_Slippy_Map)/Widget - Similar to https://tiles.stadtnavi.eu/widget/. 

The idea here is to make a map tool that at least for all *.[kommune.no](http://kommune.no/) (and [buskerudbyen.no](http://buskerudbyen.no/) [kolumbus.no](http://kolumbus.no/), [brakar.no](http://brakar.no/), banenor.no, etc.) domains are free to use. And that promotes the use of walking, cycling and public transport. And not private car usage like a lot of maps do today. (Side note: Unsure about the need to limit usage and load on servers here.)

So, it is easy to show a location - with a pin - and how to get from any starting point to the pinned location from the embedded map. For example, one button to get info on travel with public transport (using Entur's tools here) or going by bike (using Cycling Norway tools) – and maybe even with a combination of private bike and public transport (utilizing Sykkelhotell and other bicycle parking). The map should be visually appealing and have the possibility to show:
1) A pin for the location in question, with – hopefully – some additional information on the closest public transportation stop(s).
2) Some standard buttons to make it full size, zoom, etc. (Similar to uMap, and perfect if some of the design aspects could be customizable here.)
3) A button to open to and from search for cycling where the destination is already filled in. How to handle size of the map, compared to be able to show details of the route.
4) A button with Entur services. In the most basic configuration a from field to search, and when this is added a new tab in the browser opens with [standard(?) Entur search like this](https://entur.no/reiseresultater?transportModes=rail%2Ctram%2Cbus%2Ccoach%2Cwater%2Ccar_ferry%2Cmetro%2Cflytog%2Cflybuss&date=1679059038993&tripMode=oneway&walkSpeed=1.4&minimumTransferTime=120&timepickerMode=departAfter&startId=KVE%3ATopographicPlace%3A3005-Glassbl%C3%A5serveien&startLabel=Glassbl%C3%A5serveien&startLat=59.71934755553268&startLon=10.26072049905552&stopId=133185992&stopLabel=Daleg%C3%A5rdsveien%2074&stopLat=59.761859239950965&stopLon=10.125492793159967 
). The to-field is hidden or automatically populated with the address of the place we want to travel to. 

**Goal**: To replace things like this:

**Scenario 1: Maps to a specific location**
https://www.drammen.kommune.no/tjenester/barnehage/barnehagene-i-drammen/aronslokka-barnehage/ and https://www.banenor.no/reise-og-trafikk/stasjoner/-H-/Holmestrand/
And have a service that is easy to use and embed location for libraries, public swimming pools, beaches, schools, etc.

See GUI mockup number one under.

**Scenario 2: Maps that show road works for cyclists and pedestrians**
As an alternative to making images and manual work for informing travelers.
1. Step one is to make OSM data visible in the Cycling Norway solution, by using e.g. the `construction=cycleway` tag. This has to be done at least 24 hours before the change in order to be visible in the solution (and Entur).
1. In this way any route OTP will also not route over the work. 
1. Additional steps is to make a dedicated web UI for just these kinds of changes. This makes it then easier to change status, and no need for OSM Id or JOSM knowledge is needed.
1. For this over it is also possible to automatically revert the changes as well, so only one registration is needed. If the known date when the work is finished. 
To make something to replace this: https://www.stavanger.kommune.no/vei-og-trafikk/stavanger-pa-sykkel/endringer-i-sykkelruter/

See GUI mockup number two under.

**Scenario 3: Promote new cycle infrastructure and solutions**
Supported today. As long as the solution is added to OSM at least 24 hours before.

### PR DESCRIPTION
**Pull request by michaeljohansen**
A quite large PR for this project. The PR adds a new folder `widget/` with it's own bundler. The widget can be built and is dropped into `widget/dist/widget.js`. The widget can be used with a regular `<script>` tag.

**New features**
- Widget demo page that helps users get the code they need to implement the widget on their own website.
- Separate Menu component for the website and for the widget.
- Widget has pre-defined destination.
- In widget mode there is a button to start the trip from your current location.
- Routing results panel shows link to Entur with your trip details.
- Attribution panel shows "open full page" button to open the app directly from the widget.

**Cleanup and refactoring**
- Styling of a lot of components improved and made to look more unified.
- Reorganize files into folders `src/components/menu/` and `src/components/popup/`.
- Removes some old unused assets.
- Removes dependency `mapbox-gl`.
- Quite a bit of refactoring.

Resolves #27

### CODE REVIEW

File: src/components/Map.tsx [hunk 1]
```diff
@@ -69,15 +86,23 @@ const MapContainer = () => {
...
-      const to = parseLngLat(url.searchParams.get("to")!);
-      getQuery(from, to);
-    } else if (url.searchParams.has("from")) {
-      const from = parseLngLat(url.searchParams.get("from")!);
-      updateQueryFromParam(from);
+    if (props.isWidget) {
+      if (props.isWidget && url.searchParams.has("from") && dest) {
+        const from = parseLngLat(url.searchParams.get("from")!);
+        getQuery(from, dest);
+      } else if (url.searchParams.has("from")) {
```
Comment (Beck-berry) [inline src/components/Map.tsx:93]: Is this a valid case in widget mode? That there is no destionation set?

File: src/components/menu/RoutingResults.tsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+              }}
+            />
+          </div>
+          <span className="routing-results-display-text routing-results-clickable-text">
+            {elevation} m
+          </span>
+        </div>
+      </div>
+      {props.start &&
+        props.start !== null &&
```
Comment (Beck-berry) [inline src/components/menu/RoutingResults.tsx:134]: Are these both necessary?

Comment (michaeljohansen) [reply src/components/menu/RoutingResults.tsx:134]: Hah, no, well spotted.

File: src/components/Map.tsx (hunk 1 above)
Comment (michaeljohansen) [reply src/components/Map.tsx:93]: Hmm, I guess not. Removing it.

**Pull request merged**